### PR TITLE
poetry 1.2.0

### PR DESCRIPTION
**Pull request by branchv**
Created with `brew bump-formula-pr`.

### CODE REVIEW

File: Formula/poetry.rb [hunk 1]
```diff
@@ -75,6 +85,11 @@ class Poetry < Formula
     sha256 "9d643ff0a55b762d5cdb124b8eaa99c66322e2157b69160bc32796e824360e6d"
   end
 
+  resource "jsonschema" do
+    url "https://files.pythonhosted.org/packages/35/ee/889aee424a43066a06ac68e499335877775eac9b4409f7860f6af94c6688/jsonschema-4.14.0.tar.gz"
+    sha256 "15062f4cc6f591400cd528d2c355f2cfa6a57e44c820dc783aee5e23d36a831f"
+  end
```
Comment (SMillerDev) [inline Formula/poetry.rb:91]: Can this use the formula?

Comment (kozlek) [reply Formula/poetry.rb:91]: Poetry requires a specific version of `jsonschema`. As of today, Poetry requires the same version which Homebrew ships. 
However, that may not be the case tomorrow and upgrading Homebrew's `jsonschema` might break Poetry.

Comment (SMillerDev) [reply Formula/poetry.rb:91]: That's what we have tests for

Comment (kozlek) [reply Formula/poetry.rb:91]: I don't see the point of testing an upgraded dependency *on Homebrew side*.
Maintainers of Poetry's project know better than us which dependency's versions work or not.

I believe `brew update-python-resources` script is using the `pip` dependencies resolver to select which resources (and versions) to install. This behaviour ensures the resulting install will be as close as possible of a regular install using `pip install poetry` (which is done by the official poetry installer).

This way, formula's maintenance will be much easier as we will only have to take care of python's version requirements, which is much broader.

Don't get me wrong, I see your point about reusing an existing formula on the homebrew ecosystem. It avoids downloading twice or more the same dependency and that's important to save disk space (and memory in some case) but there is a tradeoff with maintainability 🙂

Comment (branchvincent) [reply Formula/poetry.rb:91]: We could, but `jsonschema` isn't currently installed globally in the same way that `six` is. So we'd have to manually add a `.pth` ourselves. Tbh, I'm not sure it's worth it here

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**